packages/perseus-editor/src/components/__tests__/locked-function-settings.test.tsx

```diff
@@ -0,0 +1,365 @@
+                    onToggle={onToggle}
+                />,
+                {
+                    wrapper: RenderStateRoot,
+                },
+            );
+
+            // Act
+            const header = screen.getByRole("button", {
+                name: "Function (y=x^2) grayH, solid",
+            });
+            await userEvent.click(header);
+
+            // Assert
+            expect(onToggle).toHaveBeenCalled();
+        });
+    });
+
+    describe("Settings interactions", () => {
+        test("calls 'onChangeProps' when color is changed", async () => {
+            // Arrange
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the color
+            const colorSwitch = screen.getByLabelText("color");
+            await userEvent.click(colorSwitch);
+            const colorOption = screen.getByText("green");
+            await userEvent.click(colorOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({color: "green"});
+        });
+
+        test("calls 'onChangeProps' when stroke style is changed", async () => {
+            // Arrange
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the stroke
+            const strokeSwitch = screen.getByLabelText("stroke");
+            await userEvent.click(strokeSwitch);
+            const strokeOption = screen.getByText("dashed");
+            await userEvent.click(strokeOption);
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledWith({strokeStyle: "dashed"});
+        });
+
+        test("calls 'onChangeProps' when equation is changed (as keys are pressed)", async () => {
+            // Arrange
+            render(
+                <LockedFunctionSettings
+                    {...defaultProps}
+                    expanded={true}
+                    onChangeProps={onChangeProps}
+                    equation=""
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Act
+            // Change the equation
+            const equationField = screen.getByLabelText("equation");
+            await userEvent.type(equationField, "zot");
+
+            // Assert
+            expect(onChangeProps).toHaveBeenCalledTimes(3);
+            expect(onChangeProps).toHaveBeenNthCalledWith(1, {equation: "z"});
+            expect(onChangeProps).toHaveBeenNthCalledWith(2, {equation: "o"});
+            expect(onChangeProps).toHaveBeenNthCalledWith(3, {equation: "t"});
```

Comment:
I'm a little confused about this. How come it's not

```
expect(onChangeProps).toHaveBeenNthCalledWith(1, {equation: "z"});
expect(onChangeProps).toHaveBeenNthCalledWith(2, {equation: "zo"});
expect(onChangeProps).toHaveBeenNthCalledWith(3, {equation: "zot"});
```
?

Reply:
I added a note in the test explaining this. The reason is due to `onChangeProps` being mocked, and not passing the typed letter to the equation field.

Reply:
Thank you!